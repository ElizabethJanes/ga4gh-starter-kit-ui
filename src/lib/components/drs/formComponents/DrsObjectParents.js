import React from 'react';
import DrsObjectRelatives from './DrsObjectRelatives';
import { 
    Typography
} from '@material-ui/core';

const DrsObjectParents = props => {
    const assertParentIsBundleThenSetValid = (responseData) => {
        if (responseData.is_bundle === true) {
            props.setParentValid();
            props.setParentName(responseData.name);
        } else {
            props.setParentInvalid();
        }
    }

    return (
        <DrsObjectRelatives
            header='Parent Bundles'
            sectionDescription={
                <div>
                    <Typography variant='body2' align='left' color='textSecondary'>
                        The following listing displays all "Parent" DRS Bundles,
                        that is, all bundles that contain the current DRS Object as
                        one of its Children.
                    </Typography>
                </div>
            }
            relationship="parents"
            objectName="parent"
            relatives={props.drs_object_parents}
            addRelative={props.addParent}
            setRelativeId={props.setParentId}
            setRelativeName={props.setParentName}
            setRelativeValid={props.setParentValid}
            setRelativeInvalid={props.setParentInvalid}
            unsetRelativeValidity={props.unsetParentValidity}
            customApiCallSuccessCallback={assertParentIsBundleThenSetValid}
            removeRelative={props.removeParent}
            readOnly={props.readOnly}
        />
    )
}

export default DrsObjectParents;
