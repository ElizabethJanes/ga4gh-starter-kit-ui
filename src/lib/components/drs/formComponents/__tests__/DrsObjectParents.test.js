import React from 'react';
import renderer from 'react-test-renderer';
import DrsObjectParents from '../DrsObjectParents';

let zeroDrsObjectParents=[];

test('SHOW <DrsObjectParents /> should handle zero child objects', () => {
    const drsObjectParents = renderer
    .create(<DrsObjectParents 
    readOnly={true} 
    formViewType='SHOW'
    drs_object_parents={zeroDrsObjectParents} />)
    .toJSON();
    expect(drsObjectParents).toMatchSnapshot();
});

test('NEW <DrsObjectParents /> should handle zero child objects', () => {
    const drsObjectParents = renderer
    .create(<DrsObjectParents 
    readOnly={false} 
    formViewType='NEW'
    drs_object_parents={zeroDrsObjectParents} />)
    .toJSON();
    expect(drsObjectParents).toMatchSnapshot();
});

test('EDIT <DrsObjectParents /> should handle zero child objects', () => {
    const drsObjectParents = renderer
    .create(<DrsObjectParents 
    readOnly={false} 
    formViewType='EDIT'
    drs_object_parents={zeroDrsObjectParents} />)
    .toJSON();
    expect(drsObjectParents).toMatchSnapshot();
});