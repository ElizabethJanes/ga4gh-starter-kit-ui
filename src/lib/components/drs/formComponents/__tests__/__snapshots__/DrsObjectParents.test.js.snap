// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EDIT <DrsObjectParents /> should handle zero child objects 1`] = `
<div
  className="MuiFormGroup-root"
>
  <div
    className="MuiBox-root MuiBox-root-3"
  >
    <hr
      className="MuiDivider-root"
    />
  </div>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft"
  >
    Parent Bundles
  </h6>
  <div>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignLeft"
    >
      The following listing displays all "Parent" DRS Bundles, that is, all bundles that contain the current DRS Object as one of its Children.
    </p>
  </div>
  <div>
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          aria-describedby={null}
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="Add parent"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`NEW <DrsObjectParents /> should handle zero child objects 1`] = `
<div
  className="MuiFormGroup-root"
>
  <div
    className="MuiBox-root MuiBox-root-2"
  >
    <hr
      className="MuiDivider-root"
    />
  </div>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft"
  >
    Parent Bundles
  </h6>
  <div>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignLeft"
    >
      The following listing displays all "Parent" DRS Bundles, that is, all bundles that contain the current DRS Object as one of its Children.
    </p>
  </div>
  <div>
    <div
      className="MuiGrid-root MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          aria-describedby={null}
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          title="Add parent"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`SHOW <DrsObjectParents /> should handle zero child objects 1`] = `
<div
  className="MuiFormGroup-root"
>
  <div
    className="MuiBox-root MuiBox-root-1"
  >
    <hr
      className="MuiDivider-root"
    />
  </div>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-alignLeft"
  >
    Parent Bundles
  </h6>
  <div>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignLeft"
    >
      The following listing displays all "Parent" DRS Bundles, that is, all bundles that contain the current DRS Object as one of its Children.
    </p>
  </div>
  <div />
</div>
`;
